pkg/operator/controllers/guardrails/guardrails_controller.go

```diff
@@ -122,24 +127,38 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		// start a ticker to re-enforce gatekeeper policies periodically
 		r.startTicker(ctx, instance)
 	} else if strings.EqualFold(managed, "false") {
+		if !r.cleanupNeeded {
+			if ns, err := r.getGatekeeperDeployedNs(ctx, instance); err == nil && ns != "" {
+				// resources were *not* created by guardrails, plus another gatekeeper deployed
+				//
+				// guardrails didn't get deployed most likely due to another gatekeeper is deployed by customer
+				// this is to avoid the accidental deletion of gatekeeper CRDs that were deployed by customer
+				// the unnamespaced gatekeeper CRDs were possibly created by a customised gatekeeper, hence cannot ramdomly delete them.
+				r.log.Warn("Skipping cleanup as it is not safe and may destroy customer's gatekeeper resources")
+				return reconcile.Result{}, nil
+			}
+		}
```

Comment:
I think we can combine the if statement into 1

Reply:
tried that, but end up making code either less readable or have to invoke the cumbersome r.getGatekeeperDeployedNs() unnecessarily